src/dev/run_check_file_casing.ts

```diff
@@ -11,12 +11,13 @@ import globby from 'globby';
 import { REPO_ROOT } from '@kbn/utils';
 import { run } from '@kbn/dev-utils';
 import { File } from './file';
+// @ts-expect-error precommit hooks aren't migrated to TypeScript yet.
 import { checkFileCasing } from './precommit_hook/check_file_casing';
 
 run(async ({ log }) => {
   const paths = await globby('**/*', {
     cwd: REPO_ROOT,
-    nodir: true,
```

Comment:
__note:__ that was an excuse to convert this file to TypeScript (this option is not supported since 8.0.0 I believe). Everything worked as expected since `onlyFiles` is set to `true` by default, but I still wanted to make it explicit here.